src/main/java/wellnus/focus/feature/FocusUi.java

```diff
@@ -0,0 +1,144 @@
+     * as the Countdown will automatically print a cursor on completion.
+     *
+     * @param session the session containing all the countdowns
+     * @param scanner the scanner reading user input
+     * @return String User input command with leading/dangling whitespace being removed
+     */
+    private String waitAndGetCommand(Session session, Scanner scanner) {
+        waitForCountdownPrint(session);
+        flushInput(scanner);
+        // Get command without printing a cursor
+        return getCommandWithCursor(NO_PRINT_CURSOR, scanner);
+    }
+
+    /**
+     * This method is the lowest abstraction layer, talking to
+     *
+     * @param isPrintCursor the session containing all the countdowns
+     * @param scanner       the scanner reading user input
+     * @return User input command with leading/dangling whitespace being removed
+     */
+    private String getCommandWithCursor(boolean isPrintCursor, Scanner scanner) {
+        String userCommand = "";
+        if (isPrintCursor) {
+            printCursor();
+        }
+        try {
+            String inputLine = scanner.nextLine();
+            userCommand = inputLine.trim();
+        } catch (BufferOverflowException bufferOverFlowException) {
+            LOGGER.log(Level.INFO, BUFFER_OVERFLOW_MSG);
+            printErrorFor(bufferOverFlowException, BUFFER_OVERFLOW_MSG);
+        } catch (NoSuchElementException noElementException) {
+            LOGGER.log(Level.INFO, NO_INPUT_ELEMENT_MSG);
+            printErrorFor(noElementException, NO_INPUT_ELEMENT_MSG);
+        }
+        return userCommand;
+    }
+
+    /**
+     * Checks if the countTimer is in an invalid phase which
+     * means that all stdin should be blocked.
+     *
+     * @param session the session containing all the countdowns
+     * @return boolean Representing the countdown state
+     */
+    public boolean isBlocking(Session session) {
+        return session.getCurrentCountdown().isCountdownPrinting() && session.isSessionCounting();
+    }
+
+    /**
+     * If the countdown is in its printing phase, wait for it to finish first
+     * The while loop is designed as such for checkstyle correctness (no empty body)
+     */
+    private void waitForCountdownPrint(Session session) {
+        while (true) {
+            if (!(isBlocking(session))) {
+                break;
+            }
+        }
+    }
+
+    /**
+     * Flush all input from Stdin.
+     * <p>
+     * Gracefully rejects all overzealous input from users, cleaning the input buffer.
+     *
+     * @param scanner Reference to the scanner being used by the UI
+     */
+    private void flushInput(Scanner scanner) {
+        // Implement a timeout to avoid hanging
+        long flushStartTime = System.currentTimeMillis();
+        while (scanner.hasNextLine()
+                && System.currentTimeMillis() - flushStartTime < 2000) {
+            // Discard extraneous input
+            scanner.nextLine();
```

Comment:
Nice that you found a way to flush malicious user input during countdown. Can I check why the 2000ms is it just to have a time greater than 1000ms because of the printing duration?

Reply:
This is arbitrary and seemed to be stable when I tested. I've refactored this magic literal out and changed the value to `1000ms` which makes more intuitive sense since each print is in 1000ms interval